head/src/main/kotlin/io/qalipsis/core/head/jdbc/repository/ZoneRepository.kt

```diff
@@ -33,4 +34,15 @@ import io.qalipsis.core.head.jdbc.entity.ZoneEntity
  */
 @R2dbcRepository(dialect = Dialect.POSTGRES)
 @Requires(notEnv = [ExecutionEnvironments.TRANSIENT])
-interface ZoneRepository : CoroutineCrudRepository<ZoneEntity, Long>
\ No newline at end of file
+interface ZoneRepository : CoroutineCrudRepository<ZoneEntity, Long> {
+
+    @Query(
+        value = """SELECT * FROM zone z
```

Comment:
This query might work (even if I'm do not understand how zones without any join in zone_tenant are fetched)  but is extremely expensive.
```
SELECT * FROM zone z
            WHERE z.enabled = true 
            AND (
               (EXISTS SELECT 1 from zone_tenant zt where z.id = zt.zone_id AND EXIST (SELECT 1 FROM tenant t WHERE t.id = zt.tenant_id AND t.reference = :tenant))
               OR
              NOT EXISTS (SELECT 1 from zone_tenant zt where z.id = zt.zone_id)
             )
```